chat-components/src/services/BroadcastService.ts

```diff
@@ -14,8 +14,66 @@ const broadcastServiceSubList: Record<string, any> = {};
 let pubChannel: any;
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 let subChannel: any;
+let eventQueue: EventQueue;
+
+class EventQueue {
+    private queueing: boolean = true;
+    private channelEventQueue: ICustomEvent[];
+    private queuingId?: NodeJS.Timeout;
+
+    constructor() {
+        this.channelEventQueue = [];
+        this.queueEvents();
+    }
+
+    processEvents() {
+        while (this.channelEventQueue.length > 0) {
+            const event = this.channelEventQueue.shift();
+            if (event) {
+                pubChannel.postMessage(event);
+            }
+        }
+    }
+
+    queueEvents(timeout = 3000) {
+        this.processEvents();
+        if (this.queueing) {
+            this.queuingId = setTimeout(() => {
+                this.queueEvents();
+            }, timeout);
+        }
+    }
```

Comment:
I suggest that instead of waiting 3 secs, to react immediately with 3 retries, 500 ms between retries, 


this aggressive approach aims to immediate notification in this fallback mechanism